src/ducks/SANCharts/ChartPage.js

```diff
@@ -23,30 +35,98 @@ class ChartPage extends Component {
     }
   }
 
-  onZoom = ({ refAreaLeft, refAreaRight }) => {
+  onZoom = (leftZoomIndex, rightZoomIndex, leftZoomDate, rightZoomDate) => {
     this.setState(
-      prevState => ({
-        ...prevState,
-        timeRange: undefined,
-        from: refAreaLeft,
-        to: refAreaRight
-      }),
-      () => {
-        this.updateSearchQuery(this.state)
-      }
+      {
+        zoom: [leftZoomIndex, rightZoomIndex + 1, leftZoomDate, rightZoomDate]
+      },
+      this.updateSearchQuery
+    )
+  }
+
+  onZoomOut = () => {
+    this.setState({ zoom: undefined }, this.updateSearchQuery)
+  }
+
+  onTimerangeChange = timeRange => {
+    this.setState({ timeRange }, this.updateSearchQuery)
+  }
+
+  onSlugSelect = ({ slug, name, ticker }) => {
+    this.setState(
+      { slug, title: `${name} (${ticker})` },
+      this.updateSearchQuery
+    )
+  }
+
+  onMetricsChange = metrics => {
+    this.setState({ metrics }, this.updateSearchQuery)
+  }
+
+  onNightModeSelect = () => {
+    const { nightMode } = this.state
+
+    this.setState(
+      { nightMode: nightMode ? undefined : true },
```

Comment:
@DmitriVanGuard Maybe smth diff?

Reply:
I agree it looks a little bit ugly and straightforward, but it is the easiest way to hide it from the querystring.